controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -534,6 +534,28 @@ func (r *BareMetalHostReconciler) actionDeleting(prov provisioner.Provisioner, i
 		return actionError{errors.Wrap(err, "failed to remove finalizer")}
 	}
 
+	secretManager := secretutils.NewSecretManager(info.log, r.Client, r.APIReader)
+
+	err = secretManager.ReleaseSecret(info.bmcCredsSecret)
+	if err != nil {
+		return actionError{err}
+	}
+
+	err = secretManager.ReleaseSecretByRef(info.host.Spec.UserData)
+	if err != nil {
+		return actionError{err}
+	}
+
+	err = secretManager.ReleaseSecretByRef(info.host.Spec.NetworkData)
+	if err != nil {
+		return actionError{err}
+	}
+
+	err = secretManager.ReleaseSecretByRef(info.host.Spec.MetaData)
+	if err != nil {
+		return actionError{err}
+	}
+
```

Comment:
I think this part could deserve a test, to verify that the finalizer is removed on host deletions from the secrets

Reply:
I assume that the proper way of adding such a test is to add it into ReleaseSecretByRef and into ReleaseSecret

Reply:
For that I was thinking more on enhancing somehow the [TestDeleteHost](https://github.com/metal3-io/baremetal-operator/blob/2de940e5f2c0603495f039e90123ddaa0f549e60/controllers/metal3.io/baremetalhost_controller_test.go#L1315) with the new cases on secrets